### PR TITLE
 daemon: Dump systemctl status rpm-ostreed on load failure

### PR DESCRIPTION
**Pull request by cgwalters**
daemon: Rename `os` variable to `hostos`

Prep for a future patch; `os` conflicts with the standard Go package.

---

daemon: Dump systemctl status rpm-ostreed on load failure

This is yet another patch which should help debugging
https://bugzilla.redhat.com/show_bug.cgi?id=1958812
like situations in the future.

I also have https://github.com/coreos/rpm-ostree/pull/2932
cooking but it's going to take a while to cycle down.

---

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -201,19 +201,25 @@ func New(
...
 	}
 
 	// Only pull the osImageURL from OSTree when we are on RHCOS or FCOS
-	if os.IsCoreOSVariant() {
+	if hostos.IsCoreOSVariant() {
 		osImageURL, osVersion, err = nodeUpdaterClient.GetBootedOSImageURL()
 		if err != nil {
+			// If this fails for some reason, let's dump the unit status
+			// into our logs to aid future debugging.
+			cmd := exec.Command("systemctl", "status", "rpm-ostreed")
```
Comment (sinnykumari) [inline pkg/daemon/daemon.go:219]: Does printing status will capture enough error logs or should we print here rpm-ostreed journal log from current boot?

Comment (cgwalters) [reply pkg/daemon/daemon.go:219]: Status will capture the last few lines of logs at the point in time this happens, which I think will be enough.  

https://github.com/openshift/must-gather/pull/244 will help for the general case.

**Review by sinnykumari — APPROVED** [review]
/lgtm

**Comment by cgwalters** [comment]
just to xref, https://github.com/coreos/rpm-ostree/pull/2945 is driving this down into rpm-ostree, so eventually we'll want to remove this code.  But right now, shipping *this* PR involved literally nothing more than me submitting it and having it merge, but changing rpm-ostree in RHEL will take much longer and involve a *lot* more exciting paperwork.

**Pull request merged**